scripts/cdn/publish.js

```diff
@@ -3,82 +3,132 @@
-let version = require('../../package.json').version;
+    if (!version || !semver.valid(version)) {
+        throw Error('package.json version is not valid');
+    }
 
-if (!version || !semver.valid(version)) {
-    throw Error('package.json version is not valid');
+    return version;
 }
 
-// Publish to CDN
-// E.g.
-//   v1.2.3: v1 / v1.2 / v1.2.3
-//   v1.2.3-beta.4: v1.2.3-beta / v1.2.3-beta.4
+function publishToCDN (version) {
+    configureS3Connection();
 
-let s3 = require('s3');
+    // Publish to CDN
+    // E.g.
+    //   v1.2.3: v1 / v1.2 / v1.2.3
+    //   v1.2.3-beta.4: v1.2.3-beta / v1.2.3-beta.4
+    let major = semver.major(version);
+    let minor = semver.minor(version);
+    let patch = semver.patch(version);
+    let prerelease = semver.prerelease(version);
 
-let client = s3.createClient({
-    s3Options: {
+    if (prerelease) {
+        /**
+        * Publish prerelease URLs
+        */
+        let base = 'v' + major + '.' + minor + '.' + patch + '-';
+        if (prerelease[0]) { // alpha, beta, rc
+            uploadFiles(base + prerelease[0]);
+        }
+        if (prerelease[1]) { // number
+            uploadFiles(base + prerelease[0] + '.' + prerelease[1]);
+        }
+    } else {
+        /**
+        * Publish release URLs
+        */
+        uploadFiles('v' + major);
+        uploadFiles('v' + major + '.' + minor);
+        uploadFiles('v' + major + '.' + minor + '.' + patch);
+    }
+}
+
+function configureS3Connection () {
+    AWS.config.update({
         accessKeyId: secrets.AWS_USER_S3_KEY,
         secretAccessKey: secrets.AWS_USER_S3_SECRET
-    }
-});
-
-let major = semver.major(version);
-let minor = semver.minor(version);
-let patch = semver.patch(version);
-let prerelease = semver.prerelease(version);
-
-if (prerelease) {
-    /**
-    * Publish prerelease URLs
-    */
-    let base = 'v' + major + '.' + minor + '.' + patch + '-';
-    if (prerelease[0]) { // alpha, beta, rc
-        uploadFiles(base + prerelease[0]);
-    }
-    if (prerelease[1]) { // number
-        uploadFiles(base + prerelease[0] + '.' + prerelease[1]);
-    }
-} else {
-    /**
-    * Publish release URLs
-    */
-    uploadFiles('v' + major);
-    uploadFiles('v' + major + '.' + minor);
-    uploadFiles('v' + major + '.' + minor + '.' + patch);
+    }, true);
+
+    S3 = new AWS.S3();
 }
 
-function uploadFiles (version) {
+async function uploadFiles (version) {
     console.log('Publish', version);
-    let uploader = client.uploadDir({
-        localDir: 'dist',
-        s3Params: {
-            ACL: 'public-read',
-            Bucket: secrets.AWS_S3_BUCKET,
-            Prefix: 'carto-vl/' + version + '/'
+
+    let files;
+    try {
+        files = await readdir('dist');
+    } catch (e) {
+        console.error(e);
+        return;
+    }
+
+    const uploadExtensions = ['.JS', '.MAP'];
+    for (const fileName of files) {
+        const extension = path.extname(fileName).toUpperCase();
```

Comment:
I'm curious, why do you go with uppercase?

Reply:
I was going to say "just in case... 😄 " but it's true, we don't need it, as we control the naming when generating the bundles.